### PR TITLE
[Bugfix][Ascend] fix npu 310/a5 bugs

### PR DESCRIPTION
**Pull request by zyz111222**
## Purpose
This PR aims to fix the bugs in running Qwen3-TTS on the Atlas A5 machine and 310.
The changes strive to minimize alterations to the mainline logic, adding A5-specific patches only where necessary. The primary adjustments include reverting torch.stft to CPU execution and rolling back the format conversion of conv.

## Test Results
seed-tts-eval zh results:
image

Performance benchmarking:
A5:
image
310:
image

### CODE REVIEW

**Comment by hsliuustc0106** [comment]
@zyz111222 Good platform-specific bugfix for NPU A5 compatibility! This PR addresses critical issues preventing Qwen3-TTS from running on Atlas A5 machines. The fix includes reverting torch.stft to CPU execution and rolling back FRACTAL_Z format conversion, with comprehensive test coverage. Minimal, targeted changes that don't impact other platforms.

**Comment by tzhouam** [comment]
@vllm-omni-review-bot please review

File: vllm_omni/platforms/npu/models/qwen3_tts_code2wav.py [hunk 1]
```diff
@@ -67,6 +79,73 @@ def _patched_load_weights(self, weights):
...
+    # on A5 devices (same gate-then-import convention as the 310P patch).
+    from vllm_omni.model_executor.models.qwen3_tts import (
+        prompt_embeds_builder,
+        qwen3_tts_talker,
+    )
+
+    class _Qwen3TTSPromptEmbedsBuilderA5(prompt_embeds_builder.Qwen3TTSPromptEmbedsBuilder):
+        """Qwen3-TTS prompt-embeds builder specialized for the A5 NPU path."""
+
+        def extract_speaker_embedding(self, wav, sr):
```
Comment (FayeSpica) [inline vllm_omni/platforms/npu/models/qwen3_tts_code2wav.py]: This duplicates the `extract_speaker_embedding` implementation in the 310P patch. Could we extract the shared CPU mel frontend into a common helper and reuse it in both paths?

File: vllm_omni/platforms/npu/models/qwen3_tts.py [hunk 2]
```diff
@@ -67,6 +79,73 @@ def _patched_load_weights(self, weights):
...
+    prompt-embeds builder swap is gated on ``is_a5()`` because it changes the
+    speaker-encoder mel front-end (``torch.stft`` is unsupported there).
+    """
+    apply_qwen3_tts_code2wav_patch()
+    if not is_a5():
+        return
+    _apply_a5_prompt_embeds_builder_patch()
+
+
+def _apply_a5_prompt_embeds_builder_patch() -> None:
```
Comment (FayeSpica) [inline vllm_omni/platforms/npu/models/qwen3_tts.py:92]: `_apply_a5_prompt_embeds_builder_patch` is outside the scope implied by `qwen3_tts_code2wav.py`. Could we rename this module to `qwen3_tts.py` now that it contains patches for both Code2Wav and the prompt-embeds builder?

File: vllm_omni/platforms/npu/models/qwen3_tts_code2wav.py (hunk 1 above)
Comment (zyz111222) [reply vllm_omni/platforms/npu/models/qwen3_tts_code2wav.py]: The extract_speaker_embedding has been extracted to the public script.

File: vllm_omni/platforms/npu/models/qwen3_tts.py (hunk 2 above)
Comment (zyz111222) [reply vllm_omni/platforms/npu/models/qwen3_tts.py:92]: This module has been renamed qwen3_tts.

File: vllm_omni/platforms/npu/__init__.py [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
 
+import torch
+
 if TYPE_CHECKING:
     from vllm_omni.platforms.npu.platform import NPUOmniPlatform
 
 __all__ = ["NPUOmniPlatform"]
 
 
+def is_a5(device: torch.device | None = None) -> bool:
```
Comment (FayeSpica) [inline vllm_omni/platforms/npu/__init__.py:16]: Non-blocking: `is_a5` and `is_310p` serve the same hardware-detection responsibility but currently live at different abstraction levels. Would it be cleaner to put them together in a shared module.
```bash
platforms/npu/__init__.py
platforms/npu/_310p/__init__.py
```

**Comment by FayeSpica** [comment]
I’ve finished my review.  Comments have been addressed, no blocking concerns from my side.

@gcanlin @tzhouam PTAL when you have time. Thanks!

File: vllm_omni/platforms/npu/__init__.py (hunk 3 above)
Comment (zyz111222) [reply vllm_omni/platforms/npu/__init__.py:16]: Currently, the differences between a5 and the mainline are minor, with only one unadapted operator. Therefore, this is essentially a workaround solution. As such, it is still expected not to be too tightly coupled with the 310. Hence, I suggest keeping this in place.

**Review by tzhouam — APPROVED** [review]
LGTM
